kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -418,24 +418,21 @@ FindGlobalEgIdResult KvbAppFilter::findGlobalEventGroupId(uint64_t external_even
   // At this point, we exhausted all private entries => it has to be a public event group
   uint64_t global_eg_id;
 
+  // If all private event groups were pruned then we need to adjust current_* to point to the last private event group
+  if (private_start == 0) {
+    ConcordAssertNE(private_end, 0);
+    current_pvt_eg_id = private_end;
+    current_ext_eg_id = private_end + public_start - 1;
+  }
+
   if (external_event_group_id < current_ext_eg_id) {
-    if (current_pvt_eg_id == private_start) {
-      std::tie(global_eg_id, std::ignore) = getValueFromTagTable(kPublicEgId, external_event_group_id);
-      return {global_eg_id, true, current_pvt_eg_id - 1, external_event_group_id};
-    }
     auto num_pub_egs = current_ext_eg_id - current_pvt_eg_id;
     auto pub_eg_id = num_pub_egs - (current_ext_eg_id - external_event_group_id - 1);
     std::tie(global_eg_id, std::ignore) = getValueFromTagTable(kPublicEgId, pub_eg_id);
     return {global_eg_id, true, current_pvt_eg_id - 1, pub_eg_id};
   }
 
   ConcordAssertGT(external_event_group_id, current_ext_eg_id);
```

Comment:
Shouldn't this be `ConcordAssertGE`? Since we skip the while loop we don't check for `current_ext_eg_id == external_event_group_id` in case of pruning.